### PR TITLE
feat: migrator support table comment

### PR DESCRIPTION
**Pull request by maiqingqiang**
- [x] Do only one thing
- [x] Non breaking API changes
- [x] Tested

### What did this pull request do?

migrator support table comment.

1. [https://github.com/go-gorm/gorm/pull/6225](https://github.com/go-gorm/gorm/pull/6225)
2. [https://github.com/go-gorm/mysql/pull/114](https://github.com/go-gorm/mysql/pull/114)
3. [https://github.com/go-gorm/gen/pull/822](https://github.com/go-gorm/gen/pull/822)

### CODE REVIEW

**Comment by jinzhu** [comment]
Maybe better to have an API like `ColumnTypes` to return table-related information...

**Comment by maiqingqiang** [comment]
> Maybe better to have an API like `ColumnTypes` to return table-related information...

Okay, I'll make some changes.

**Comment by jinzhu** [comment]
> > Maybe better to have an API like `ColumnTypes` to return table-related information...
> 
> Okay, I'll make some changes.

Thank you, we can add a `Tables` method to the `Migrator` interface, it returns a `TableType` interface which contains `Name`, `Comment`... methods

**Comment by maiqingqiang** [comment]
> > > Maybe better to have an API like `ColumnTypes` to return table-related information...
> > 
> > 
> > Okay, I'll make some changes.
> 
> Thank you, we can add a `Tables` method to the `Migrator` interface, it returns a `TableType` interface which contains `Name`, `Comment`... methods

Yeah, I think so too.

image

**Comment by maiqingqiang** [comment]
I've finished developing the `TableType` method. @jinzhu 

```go
        tableType := g.db.Migrator().TableType(tableName)
	log.Printf("table catalog: %s", tableType.Catalog())
	log.Printf("table schema: %s", tableType.Schema())
	log.Printf("table name: %s", tableType.Name())
	log.Printf("table type: %s", tableType.Type())

	engine, ok := tableType.Engine()
	log.Printf("table engine: %s ok:%v", engine, ok)

	comment, ok = tableType.Comment()
	log.Printf("table comment: %s ok:%v", comment, ok)

	log.Printf("tableType: %+v", tableType)
```

image

File: migrator.go [hunk 1]
```diff
@@ -56,6 +56,16 @@ type Index interface {
 	Option() string
 }
 
+// TableType table type interface
+type TableType interface {
```
Comment (jinzhu) [inline migrator.go:60]: Hi @maiqingqiang 

Maybe we can start with the `Schema`, `Name`, `Type`, and `Comment` fields, which are less confusing for all drivers.

**Review by jinzhu — APPROVED** [review]

**Comment by yiranzai** [comment]
LGTM

**Pull request merged**